sqlmesh/dbt/target.py

```diff
@@ -257,17 +255,17 @@ class SnowflakeConfig(TargetConfig):
     retry_all: bool = False
 
     @model_validator(mode="before")
-    @model_validator_v1_args
-    def validate_authentication(
-        cls, values: t.Dict[str, t.Union[t.Tuple[str, ...], t.Optional[str], t.Dict[str, t.Any]]]
-    ) -> t.Dict[str, t.Union[t.Tuple[str, ...], t.Optional[str], t.Dict[str, t.Any]]]:
-        if (
-            values.get("password")
-            or values.get("authenticator")
-            or values.get("private_key")
-            or values.get("private_key_path")
-        ):
-            return values
+    @classmethod
+    def validate_authentication(cls, data: t.Any) -> t.Any:
+        if isinstance(data, dict):
```

Comment:
[Nit] Can this be a part of a single `if` block?

Reply:
Done